effectful/ops/syntax.py

```diff
@@ -834,13 +839,19 @@ def _(op, *args, **kwargs):
     from effectful.ops.semantics import apply, evaluate, typeof
 
     arg_ctxs, kwarg_ctxs = op.__fvs_rule__(*args, **kwargs)
+    args_, kwargs_ = list(args), dict(kwargs)
+
+    if len(args) != len(arg_ctxs):
```

Comment:
Why is this check necessary, and why is it here rather than inside the implementation of `Operation.__fvs_rule__`, for which this should be a post-condition? Was this failing before?

Reply:
Is there a test that was passing incorrectly before and now fails with the correct error? I'm confused because `__fvs_rule__` should already guarantee that this holds, so if it ever fails it means there's a bug there.

Reply:
I see. I'll remove this check then, because I don't have a test for it, and I don't remember the exact context in which it was added.